cwms-data-api/src/test/java/cwms/cda/data/dto/BasinTest.java

```diff
@@ -0,0 +1,75 @@
+package cwms.cda.data.dto;
+
+import cwms.cda.api.errors.FieldException;
+import cwms.cda.data.dto.basin.Basin;
+import cwms.cda.formatters.Formats;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+
+final class BasinTest {
+    @Test
+    void testBasinSerializationRoundTrip() {
+        Basin basin = buildTestBasin(new LocationIdentifier.Builder()
+                .withLocationId("TEST_LOCATION1")
+                .withOfficeId("NVE")
+                .build());
+        String serialized = Formats.format(Formats.parseHeader(Formats.JSONV1, Basin.class), basin);
+        Basin deserialized = Formats.parseContent(Formats.parseHeader(Formats.JSONV1, Basin.class), serialized, Basin.class);
+        assertEquals(basin, deserialized, "Roundtrip serialization failed");
+        assertEquals(basin.hashCode(), deserialized.hashCode(), "Roundtrip serialization failed");
+    }
+
+    @Test
+    void testBasinSerializationRoundTripFromFile() throws Exception {
+        Basin basin = buildTestBasin(new LocationIdentifier.Builder()
+                .withLocationId("TEST_LOCATION2")
+                .withOfficeId("MVR")
+                .build());
+        InputStream resource = this.getClass().getResourceAsStream("/cwms/cda/data/dto/basin/basin.json");
```

Comment:
Add in a 2nd basin json file where the parent basin is null (this is the typical use case)

Reply:
Addressed in latest push